common/src/lib.rs

```diff
@@ -2,9 +2,73 @@ pub mod crypto;
 
 use crypto::*;
 use serde::{Deserialize, Serialize};
-use std::collections::BTreeSet;
+use std::collections::{BTreeMap, BTreeSet};
 use thiserror::Error;
 
+/// A set of state that is directly recorded in the header.
```

Comment:
`A set of states` or `The state`?
I think the latter is what you meant but please have it checked

Reply:
fixed.